### PR TITLE
Boost: Update Settings Page to react

### PR DESCRIPTION
**Pull request by dilirity**
## Proposed changes:

* Update Settings Page to react.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pc9hqz-2jy-p2

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

* Make sure that everything on the main Settings page works as expected.

### CODE REVIEW

File: projects/plugins/boost/app/assets/src/js/layout/settings-page/settings-page.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+	useEffect( () => {
+		const unsubscribe = isGenerating.subscribe( value => {
+			setIsGeneratingValue( value );
+		} );
+
+		return () => {
+			unsubscribe();
+		};
+	}, [] );
```
Comment (haqadn) [inline projects/plugins/boost/app/assets/src/js/layout/settings-page/settings-page.tsx:30]: This is so cool! If I kinda want a custom hook now:

```javascript
const useSvelteStore = (svelteStore) => {
	const [ value, setValue ] = useState( svelteStore.get() );
	
	useEffect( () => {
		const unsubscribe = svelteStore.subscribe( value => {
			setValue( value );
		} );

		return () => {
			unsubscribe();
		};
	}, [] );

	return value;
}
```

Comment (thingalon) [reply projects/plugins/boost/app/assets/src/js/layout/settings-page/settings-page.tsx:30]: I mean if we want to rush to the end of the React migration, there is absolutely nothing magical about Svelte stores. I could write a non-svelte replacement for `writable` in a few lines, we could wrap them into React like so ^ and then we just have a React codebase to tidy up. Just sayin'.

**Review by haqadn — CHANGES_REQUESTED** [review]
The critical CSS states aren't updating as it generates. The UI get's stuck with the mounted state. I suspect it has something to do with embedding React in Svelte.

It might be worth it to ignore this issue and merge it while fixing it in my PR since it will be all React in there and we won't have to deal with the added complexity. Let me know if that's what you too think we should do.

**Review by haqadn — APPROVED** [review]
As discussed in DM, giving it a ✅  for merging as is even though the critical CSS UI is currently broken. It will be fixed in #34731

**Pull request merged**